docs/distribution/README.md

```diff
@@ -0,0 +1,222 @@
+        "architecture": "ppc64le",
+        "os": "linux"
+      }
+    }
+  ]
+}
+```
+
+## Implementation
+For illustration, let's say that an image already exists in the registry:
+- repository: `hello-world`
+- digest: `sha256:80559bf80b44ce6be8234e6ff90a1ac34acbeb826903b02cfa0da11c82cbc042`
+- tag: `v1.0`
+
+The storage layout is as follows:
+
+```
+<root>
+└── v2
+    └── repositories
+        └── hello-world
+            └── _manifests
+                └── revisions
+                    └── sha256
+                        └── 80559bf80b44ce6be8234e6ff90a1ac34acbeb826903b02cfa0da11c82cbc042
+                            └── link
+```
+
+Now we push the OCI index with the signature reference in its config. The client would first upload the signature blob as a layer, followed by the index. The digest of the index is `sha256:90659bf80b44ce6be8234e6ff90a1ac34acbeb826903b02cfa0da11c82cbc042`.
+
+```json
+{
+  "schemaVersion": 2.1,
+  "mediaType": "application/vnd.oci.image.index.v2+json",
+  "config": {
+    "mediaType": "application/vnd.cncf.notary.config.v2+jwt",
+    "digest": "sha256:2235d2d22ae5ef400769fa51c84717264cd1520ac8d93dc071374c1be49cc77c",
+    "size": 1906
+  },
+  "manifests": [
+    {
+      "mediaType": "application/vnd.oci.image.manifest.v1+json",
+      "digest": "sha256:80559bf80b44ce6be8234e6ff90a1ac34acbeb826903b02cfa0da11c82cbc042",
+      "size": 7023,
+      "platform": {
+        "architecture": "ppc64le",
+        "os": "linux"
+      }
+    }
+  ]
+}
+```
+
+At this point, the manifests storage layout will look as follows:
+
+```
+<root>
+└── v2
+    └── repositories
+        └── hello-world
+            └── _manifests
+                └── revisions
+                    └── sha256
+                        ├── 80559bf80b44ce6be8234e6ff90a1ac34acbeb826903b02cfa0da11c82cbc042
+                        │   └── link
+                        │   └── signatures
+                        │       └── sha256
+                        │           └── 2235d2d22ae5ef400769fa51c84717264cd1520ac8d93dc071374c1be49cc77c
+                        │               └── link
+                        └── 90659bf80b44ce6be8234e6ff90a1ac34acbeb826903b02cfa0da11c82cbc042
+                            └── link
+```
+
+Let's add another signature for the manifest `sha256:80559bf80b44ce6be8234e6ff90a1ac34acbeb826903b02cfa0da11c82cbc042`, where the digest of the index is `sha256:007170c33ebc4a74a0a554c86ac2b28ddf3454a5ad9cf90ea8cea9f9e75a153b`.
+
+```json
+{
+  "schemaVersion": 2.1,
+  "mediaType": "application/vnd.oci.image.index.v2+json",
+  "config": {
+    "mediaType": "application/vnd.cncf.notary.config.v2+jwt",
+    "digest": "sha256:3335d2d22ae5ef400769fa51c84717264cd1520ac8d93dc071374c1be49cc88d",
+    "size": 1906
+  },
+  "manifests": [
+    {
+      "mediaType": "application/vnd.oci.image.manifest.v1+json",
+      "digest": "sha256:80559bf80b44ce6be8234e6ff90a1ac34acbeb826903b02cfa0da11c82cbc042",
+      "size": 7023,
+      "platform": {

```

Comment:
Is platform needed? Since this isn't a multi-arch index, I was hoping we could avoid having this detail.

Reply:
I copy-pasted this from your PR to notaryproject, it doesn't have any effect here